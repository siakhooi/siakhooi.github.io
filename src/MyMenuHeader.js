import React from 'react'
import PropTypes from 'prop-types'

MyMenuHeader.propTypes = {
  selected: PropTypes.string,
  onClick: PropTypes.func,
  name: PropTypes.string
}

function MyMenuHeader (props) {
  return (
    // eslint-disable-next-line react/no-unknown-property
    <button className="MyMenuHeader" currentTab={props.selected} onClick={props.onClick}>{props.name}</button>
  )
}
export default MyMenuHeader
