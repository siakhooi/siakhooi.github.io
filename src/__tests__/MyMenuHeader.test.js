import React from 'react'
import renderer from 'react-test-renderer'
import MyMenuHeader from '../MyMenuHeader'

it('MyMenuHeader/1', () => {
  const dummyFunction = jest.fn()

  const component =
    renderer.create(
      <MyMenuHeader onClick={dummyFunction} selected='Y' name='Hello World' />
    )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onClick()
  expect(dummyFunction).toHaveBeenCalled()
})
