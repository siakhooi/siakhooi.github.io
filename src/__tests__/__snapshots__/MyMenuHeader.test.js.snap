// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyMenuHeader/1 1`] = `
<button
  className="MyMenuHeader"
  currentTab="Y"
  onClick={[MockFunction]}
>
  Hello World
</button>
`;
